app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/DirectoryFragment.kt

```diff
@@ -44,39 +41,32 @@ class DirectoryFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        viewModel = ViewModelProvider(requireActivity())[ContactsViewModel::class.java]
+        viewModel.setDatabase(db)
+        adapter = ContactsRecyclerAdapter()
+        adapter.updateContacts(viewModel.contacts.value ?: emptyList())
```

Comment:
Consider adding null safety checks for the viewModel.contacts.value property, since it may be null at certain points in the code. This would prevent NullPointerExceptions and make the code more robust.
For example:
`adapter.updateContacts(viewModel.contacts.value?.let { contacts ->
contacts
} ?: emptyList())`

Reply:
Not sure what you mean here, how would this help the program, the function is called once here and then will only be called in the event listener. Here I am already checking whether the value is null or not and I am adapting accordingly.